server/schedulers/hot_region.go

```diff
@@ -813,17 +809,27 @@ func (bs *balanceSolver) getHotDecRatioByPriorities(srcLd, dstLd *statistics.Sto
 		}
 		return a - b
 	}
-	checkHot := func(dim int) (bool, float64) {
-		srcRate := srcLd.Loads[dim]
-		dstRate := dstLd.Loads[dim]
-		peerRate := peer.GetLoad(statistics.GetRegionStatKind(bs.rwTy, dim))
-		decRatio := (dstRate + peerRate) / getSrcDecRate(srcRate, peerRate)
-		isHot := peerRate >= bs.getMinRate(dim)
-		return isHot, decRatio
-	}
-	firstHot, firstDecRatio := checkHot(bs.firstPriority)
-	secondHot, secondDecRatio := checkHot(bs.secondPriority)
-	return firstHot, firstDecRatio, secondHot, secondDecRatio
+	srcRate, dstRate := bs.cur.getExtremeLoad(dim)
+	peerRate := bs.cur.getPeerRate(bs.rwTy, dim)
+	isHot := peerRate >= bs.getMinRate(dim)
+	decRatio := (dstRate + peerRate) / getSrcDecRate(srcRate, peerRate)
+	return isHot, decRatio
+}
+
+func (bs *balanceSolver) isBetterForWriteLeader() bool {
+	srcRate, dstRate := bs.cur.getExtremeLoad(bs.firstPriority)
+	peerRate := bs.cur.getPeerRate(bs.rwTy, bs.firstPriority)
+	return srcRate-peerRate >= dstRate+peerRate && bs.isTolerance(bs.firstPriority)
+}
+
+func (bs *balanceSolver) isBetter(dim int) bool {
+	isHot, decRatio := bs.getHotDecRatioByPriorities(dim)
+	return isHot && decRatio <= bs.sche.conf.GetGreatDecRatio() && bs.isTolerance(dim)
+}
+
+func (bs *balanceSolver) isNotWorsened(dim int) bool {
+	_, decRatio := bs.getHotDecRatioByPriorities(dim)
+	return decRatio <= bs.sche.conf.GetMinorDecRatio()
```

Comment:
```suggestion
	isHot, decRatio := bs.getHotDecRatioByPriorities(dim)
	return !isHot || decRatio <= bs.sche.conf.GetMinorDecRatio()
```

It's just a discussion, is it necessary for us to make such changes in the future?